### PR TITLE
[doc] Fix typos and inline links

### PR DESCRIPTION
**Pull request by Bharat123rox**
*Description of changes:*
Fixes some inline RST links and typos in the tutorial docs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-609-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-609/1/index.html

File: docs/tutorials/image_classification/beginner.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ filename = ag.download('https://autogluon.s3.amazonaws.com/datasets/shopee-iet.z
 ag.unzip(filename)
 ```
 
-After the dataset is downloaded, we load it into a [`Dataset`](/api/autogluon.task.html#autogluon.task.ImageClassification.Dataset) object: 
+After the dataset is downloaded, we load it into a :class:`Dataset` object:
```
Comment (zhreshold) [inline docs/tutorials/image_classification/beginner.md:28]: after checking the rendered website, the :class: sphinx ref is not working.

Comment (Innixma) [reply docs/tutorials/image_classification/beginner.md:28]: Do you have a suggestion to render it properly?

Comment (Bharat123rox) [reply docs/tutorials/image_classification/beginner.md:28]: I think the solution would be to revert to the hyperlink and get rid of the "code" style as Sphinx doesn't support hyperlinks in a code style to the best of my knowledge @zhreshold @innixma

I'll make the changes in a few hours

**Comment by Bharat123rox** [comment]
Looks to be an intermittent `conda`/`pip` error, can someone look into that and retrigger the build if possible? Thanks

**Comment by zhreshold** [comment]
@Bharat123rox Can you rebase against the master branch? The CI issue has been fixed

**Comment by szha** [comment]
Job PR-609-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-609/3/index.html

**Comment by zhreshold** [comment]
@Bharat123rox Thanks! This is merged

**Pull request merged**